Assets/script/ShopController.cs

```diff
@@ -57,31 +59,31 @@ void Update()
         this.shojicoin.GetComponent<Text>().text = "所持コイン " + GameDirector.savecoin;
```

Comment:
"所持コイン " + GameDirector.savecoin

こういう書き方にすることでスマートになります。

$"所持コイン {GameDirector.savecoin}"

Reply:
対応しました！

Reply:
「$」が抜けてます。


string text = "あいうえお";
int num = 123;
①　”平仮名: ” + text + " / " + "数字: " + num.ToString();
②　string.Format(平仮名: {0} / 数字: {1}, text, num);
③　$"平仮名: {text} / 数字: {num}";

この３つの記述はどれも同じ結果になります。

複数の変数や数値が混ざった文字列を扱う場合、
③が一番スマートに書けるので覚えておきましょう。
現場でよく使います。

Reply:
すいません。対応しました。

Reply:
$の場所違います。
ちゃんとゲーム再生して確認してる？
表示おかしくなってるはずだよ。

Reply:
$を指定された通りにするとこのようなエラーが出ます。
Assets/Scripts/ShopController.cs(59,32): error CS1644: Feature `interpolated strings' cannot be used because it is not part of the C# 4.0 language specification

エラーの内容をプッシュしました。

Reply:
ググりましょう。

http://bluebirdofoz.hatenablog.com/entry/2017/12/27/004737

上記の記事に同じような現象と解決法が書いてありました。
この対処を試すか、Unityのバージョンアップをするかです。
2017はもう古い部類になってきてますので、今回のように書けない記述の仕方が出てきています。